src/iuse_actor.cpp

```diff
@@ -1078,10 +1078,12 @@ cata::optional<int> deploy_furn_actor::use( Character &p, item &it, bool,
         p.add_msg_if_player( m_info, _( "There is already furniture at that location." ) );
         return cata::nullopt;
     }
-    if( here.has_items( pnt ) ) {
-        p.add_msg_if_player( m_info, _( "Before deploying furniture, you need to clear the tile." ) );
-        return cata::nullopt;
+    
+    if( here.mop_spills( tripoint_bub_ms( pnt ) ) ) {
+        p.add_msg_if_player( m_info, _( "You moped up the spill when deploying furniture." ) );
+        p.moves -= 35;
     }
```

Comment:
You need to add check for a mop in player's possession. Otherwise liquids will be mopped by magic, without any mop.

Reply:
Can I avoid checking the mop if I change the message to "The liquid is slowly evaporating"? Or just clean up the water without generating a prompt at all.

Reply:
I was going to simulate the player using their hands or feet to clear the water away by increasing the time consumed by the action(The original time consumed for mopping was 15, and I changed it to 35).

Reply:
After all, in the real world, water evaporates and there is currently no such mechanism in the game.

Reply:
I have now replaced the prompt message.

Reply:
Detection of mops has now been added, allowing you to mop up liquids when deploying furniture as long as the mop is within a certain radius of the player.